File: TabLayout/app/src/main/java/com/example/tablayout/MainActivity.java
package com.example.tablayout;

import android.content.Context;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    TabLayout tabLayout;
    ViewPager viewPager;
    Halaman adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tabLayout=findViewById(R.id.tab);
        viewPager=findViewById(R.id.viewpager);

        adapter=new Halaman(getApplicationContext(),getSupportFragmentManager(),tabLayout.getTabCount());
        viewPager.setAdapter(adapter);
        viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));
        tabLayout.setOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                viewPager.setCurrentItem(tab.getPosition());
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {

            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {

            }
        });
    }
}
class Halaman extends FragmentStatePagerAdapter{
    Context context;
    int jumlah_tab;
    Halaman(Context context, FragmentManager fm,int jumlah_tab){
        super(fm);
        this.context=context;
        this.jumlah_tab=jumlah_tab;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position){
            case  0:return new Fragment1();
            case  1:return new Fragment2();
            case  2:return new Fragment3();
        }
        return null;
    }

    @Override
    public int getCount() {
        return jumlah_tab;
    }
}

File: ListVieww/app/src/main/java/com/example/listvieww/MainActivity.kt
package com.example.listvieww

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.ListView

class MainActivity : AppCompatActivity() {
    lateinit var listview:ListView;
    lateinit var nama:ArrayList<String>;
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        tambah_data();
        listview=findViewById(R.id.listview);
        var arrayAdapter=ArrayAdapter(this,android.R.layout.simple_list_item_1,nama);
        listview.adapter=arrayAdapter;
    }

    fun tambah_data(){
        nama=ArrayList<String>();
        nama.add("tento");
        nama.add("luffy");
        nama.add("Zoro");
        nama.add("sanji");
    }
}